File: pset1/mario.c
#include <cs50.h>
#include <stdio.h>

void newline(void);
void repeat_char(char c, int n);

int main(void)
{
    int height = -1;
    char block = '#';
    char space = ' ';

    // Get height of pyramid
    while (height < 0 || height > 23)
    {
        printf("Height: ");
        height = get_int();
    }
    
    // Display pyramid
    for (int i = height - 1; i >= 0; i--)
    {
        repeat_char(space, i);
        repeat_char(block, height - i);
        
        repeat_char(space, 2);
        
        repeat_char(block, height - i);
        
        newline();
    }
}

void repeat_char(char c, int n)
{
    for (int i = 0; i < n; i++)
    {
        printf("%c", c);
    }
}

void newline(void)
{
    printf("\n");
}

File: pset6/mario.py
import cs50

space = ' '
block = '#'

def main():
    height = -1
    while height < 0 or 23 < height:
        height = get_height()
    
    for i in range(height):
        sp = height - i - 1
        bsp = i + 1
        
        print(space * sp,  end='')
        print(block * bsp, end='')
        print(space * 2,   end='')
        print(block * bsp, end='')
        
        print()
        
def get_height():
    print('Height: ', end='')
    height = cs50.get_int()
    return height;
        
if __name__ == '__main__':
    main()